plugins/ui/src/js/src/elements/TextArea.tsx

```diff
@@ -21,52 +18,32 @@ export function TextArea(props: TextAreaProps): JSX.Element {
     defaultValue = '',
     value: propValue,
     onChange: propOnChange = EMPTY_FUNCTION,
+    icon: propIcon,
     ...otherProps
   } = props;
 
-  const [value, setValue] = useState(propValue ?? defaultValue);
-  const [pending, setPending] = useState(false);
-  const prevPropValue = usePrevious(propValue);
-
-  // Update local value to new propValue if the server sent a new propValue and no user changes have been queued
-  if (
-    propValue !== prevPropValue &&
-    propValue !== value &&
-    propValue !== undefined &&
-    !pending
-  ) {
-    setValue(propValue);
-  }
-
-  const propDebouncedOnChange = useCallback(
-    async (newValue: string) => {
-      try {
-        await propOnChange(newValue);
-      } catch (e) {
-        log.warn('Error returned from onChange', e);
-      }
-      setPending(false);
-    },
-    [propOnChange]
-  );
-
-  const debouncedOnChange = useDebouncedCallback(
-    propDebouncedOnChange,
-    VALUE_CHANGE_DEBOUNCE
+  const [value, onChange] = useDebouncedOnChange<string>(
```

Comment:
Unnecessary to specify - TS is smart enough to infer the type correctly from the value passed in.
```suggestion
  const [value, onChange] = useDebouncedOnChange(
```